src/main/java/seedu/address/model/task/TaskDeadline.java

```diff
@@ -3,16 +3,20 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
 /**
  * Represents a Task's deadline, must only contain alphanumeric characters.
  * Guarantees: immutable; is valid as declared in {@link #isValidTaskDeadline(String)}
  */
 public class TaskDeadline {
 
-
-    public static final String MESSAGE_CONSTRAINTS =
-            "TaskDeadline must only contain alphanumeric characters and dashes";
-    public static final String VALIDATION_REGEX = "[a-zA-Z0-9-]*";
+    public static final String MESSAGE_CONSTRAINTS = "TaskDeadline must not be blank and must only consist of one \n"
```

Comment:
Nice! Will update the UG to specify that the time field is optional.

Reply:
Date is also optional! but must at least have 1 of them

Reply:
So it can be date only/time only/both date and time but tbh now that I think about it again, it doesn't really make sense to not include date